src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -419,12 +419,18 @@ SR_PRIV int ols_receive_data(int fd, int revents, void *cb_data)
 
 	received_a_byte = FALSE;
 	while (revents == G_IO_IN &&
-	       serial_read_nonblocking(serial, &byte, 1) == 1) {
+	       (num_bytes_read = serial_read_nonblocking(
+			serial, devc->raw_sample + devc->raw_sample_size,
+			num_changroups - devc->raw_sample_size)) > 0) {
 		received_a_byte = TRUE;
```

Comment:
Yes, timeout handling is unaffected. We're just reading more at the same time.

About the variable: Since a sample can span multiple bytes but the serial interface might only deliver 1 byte (theoretically), I think `received_a_byte` is actually still correct.